drivers/gpio/gpio_mchp_xec_v2.c

```diff
@@ -141,6 +144,12 @@ static int gpio_xec_configure(const struct device *dev,
 	pcr1_addr = pin_ctrl_addr(dev, pin);
 	pcr1 = sys_read32(pcr1_addr);
 
+	/* Check if pin is in GPIO mode */
+	if (MCHP_GPIO_CTRL_MUX_GET(pcr1) != MCHP_GPIO_CTRL_MUX_F0) {
+		LOG_WRN("Port:%d pin:0x%x not in GPIO mode. CTRL[%x]=%x", config->port_num, pin,
```

Comment:
I don't know this part. Are there situations where you want the pin configured while not in GPIO mode?

If not, this should be an assert since you want them to know they've messed up immediately.

Reply:
@glenn-andrews  I'm only aware of one case where PS2 data interrupt needed to be enabled via GPIO interrupt even though in native pin mode. (earlier Zephyr release limitation) That's why I thought about the log warning first to find these issues/limitations if any then once they have a path to cleaner implementations use the ASSERT